source/parser.hera

```diff
@@ -4424,18 +4428,21 @@ NestedJSXChildExpression
 ## Type Stuff
 
 TypeDeclaration
-  (TypeDeclarationModifier TrailingComment*)* TypeDeclarationRest -> { ts: true, children: $0 }
-
-TypeDeclarationModifier
-  "declare" NonIdContinue
-  Export
+  ( Export _? )? ( Declare _? )  TypeLexicalDeclaration
+  ( Export _? )? ( Declare _? )? TypeDeclarationRest -> { ts: true, children: $0 }
 
 TypeDeclarationRest
   TypeKeyword TrailingComment* IdentifierName TypeParameters? __ Equals __ Type
   Interface   TrailingComment* IdentifierName TypeParameters? InterfaceExtendsClause? InterfaceBlock
   Namespace   TrailingComment* IdentifierName NamespaceBlock
   FunctionSignature
 
+TypeLexicalDeclaration
```

Comment:
Maybe add a comment that `TypeLexicalDeclaration` is for when there's definitely a `declare` prefix so it's for TypeScript only?

Also, it seems like this is where `class` belongs.